src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/AllVersionsPropsFile.cs

```diff
@@ -36,21 +36,43 @@ public AllVersionsPropsFile(Dictionary<string, string> versions)
         Versions = versions;
     }
 
-    public (string? Sha, string? Version) GetVersion(string repository)
+    public (string? Sha, string? Version)? GetVersion(string repository)
     {
         var key = SanitizePropertyName(repository) + ShaPropertyName;
         Versions.TryGetValue(key, out var sha);
 
         key = SanitizePropertyName(repository) + PackageVersionPropertyName;
         Versions.TryGetValue(key, out var version);
+
+        if (sha is null && version is null)
+        {
+            return null;
+        }
+
         return (sha, version);
     }
 
-    public void UpdateVersion(string repository, string sha, string version)
+    public void UpdateVersion(string repository, string? sha, string? version)
```

Comment:
Should be covered by tests

Reply:
The class has tests already. Or do you mean cover the null values?

Reply:
That's right, but you are adding a null-branching logic which doesn't seem to be reflected in the tests.

Reply:
Not anymore (with the enclosing type name)

Reply:
Actually yes, for the respective items inside. I will add a test

Reply:
I have disallowed null as it's not a valid use case anyway.